yspryn/hw6/min_max.py

```diff
@@ -0,0 +1,94 @@
+def min_ysp(*args, key=None):
+    res = args[0]
+    if key:
+        for i in args:
+            if key(i) < key(res):
+                res = i
+    else:
+        for i in args:
+            if i < res:
+                res = i
+    return res
+
+
+def max_ysp(*args, key=None):
+    res = args[0]
+    if key:
+        for i in args:
+            if key(i) > abs(res):
+                res = i
+    else:
+        for i in args:
+            if i > res:
+                res = i
+    return res
+
+
+def sorted_bubbles_method(*args, key=None, reverse=False):
+    list_to_sort = list(args)
+    stop_mark = True
+    if key:
+        while stop_mark:
+            stop_mark = False
+            for i in range(len(list_to_sort) - 1):
+                if reverse:
```

Comment:
reverese never changes, 
at the begining of function:

https://docs.python.org/3/library/operator.html

```
from operator import lt, gt

compare = gt if reverse else lt
```